File: 283-move-zeroes/move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        
        int crr=0,prev=0;

        while(crr<nums.length){
            if(nums[crr]==0){
                crr++;
            }
            else{
                int temp=nums[prev];
                nums[prev]=nums[crr];
                nums[crr]=temp;

                prev++;
                crr++;
            }
        }
    }
}

File: 283-move-zeroes/move-zeroes.c
void moveZeroes(int* nums, int numsSize) {
    int prev=0,crr=0;

    while(crr<numsSize){
        if(nums[crr]!=0){
            int temp=nums[crr];
            nums[crr]=nums[prev];
            nums[prev]=temp;
            prev++;
            crr++;
        }
        else{
            crr++;
        }
    }
}